### PR TITLE
ironbank: enable /usr/bin/tini

### PR DESCRIPTION
**Pull request by v1v**
## Proposed commit message

Create symlink to `/usr/bin/tini` in the ironbank docker images.

## Why is it important?

`ironbank` images can be used in ECK.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Comment by v1v** [comment]
### Test

It's possible to validate the new docker tar context but for such we won't be able to use the official UBI image but `docker.elastic.co/ubi9/ubi-minimal:latest`.

Given the artifacts stored in the beats google storage bucket for this particular PR then:

```bash
gsutil -m -q cp -r gs://beats-ci-artifacts/beats/commits/89ac40b6f997b5ae46fb2277c6b39bf693129f0c/auditbeat/auditbeat-ironbank-8.12.0-SNAPSHOT-docker-build-context.tar.gz .
gsutil -m -q cp -r gs://beats-ci-artifacts/beats/commits/89ac40b6f997b5ae46fb2277c6b39bf693129f0c/auditbeat/auditbeat-8.12.0-SNAPSHOT-linux-x86_64.tar.gz auditbeat-8.12.0-linux-x86_64.tar.gz
tar -xvzf auditbeat-ironbank-8.12.0-SNAPSHOT-docker-build-context.tar.gz
wget https://github.com/krallin/tini/releases/download/v0.19.0/tini-amd64 -O tinit 
wget -O jq https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 

cat > Dockerfile.base << EOF
ARG BASE_REGISTRY=docker.elastic.co
ARG BASE_IMAGE=ubi9/ubi-minimal
ARG BASE_TAG=latest

FROM \${BASE_REGISTRY}/\${BASE_IMAGE}:\${BASE_TAG}
RUN microdnf install tar gzip findutils shadow-utils -y
EOF

docker build \
  -f Dockerfile-base \
  --tag docker.elastic.co/ubi9/ubi-minimal:latest-new \
  --platform linux/amd64 .

docker build --build-arg BASE_REGISTRY=docker.elastic.co \
  --build-arg BASE_IMAGE=ubi9/ubi-minimal \
  --build-arg BASE_TAG=latest-new \
  --platform linux/amd64 .
```

Then `gh repo clone elastic/ironbank`

Modify `ironbank/src/observability/auditbeat/docker-compose.yml` and use a docker image rather than building the docker image from scratch

```diff
diff --git a/src/observability/auditbeat/docker-compose.yml b/src/observability/auditbeat/docker-compose.yml
index 0eb3057..c8743e1 100644
--- a/src/observability/auditbeat/docker-compose.yml
+++ b/src/observability/auditbeat/docker-compose.yml
@@ -1,15 +1,7 @@
 version: "2.4"
 services:
   beat:
-    image: dcar-auditbeat:${STACK_VERSION}
-    build:
-      context: src
-      network: host
-      args:
-        - BASE_REGISTRY=${BASE_REGISTRY}
-        - BASE_IMAGE=${BASE_IMAGE}
-        - BASE_TAG=${BASE_TAG}
-        - STACK_VERSION=${STACK_VERSION}
+    image: ${DOCKER_IMAGE}

```

And finally run docker-compose to validate the healthy of the system

```bash
DOCKER_IMAGE=auditbeat:latest \
  STACK_VERSION=8.12.0-SNAPSHOT \
  docker-compose up --no-start

DOCKER_IMAGE=auditbeat:latest \
  STACK_VERSION=8.12.0-SNAPSHOT \
  docker-compose start

DOCKER_IMAGE=auditbeat:latest \
  STACK_VERSION=8.12.0-SNAPSHOT \
  docker-compose exec beat curl -I -f --max-time 5 --unix-socket '/usr/share/auditbeat/data/auditbeat.sock' 'http:/stats?pretty'

DOCKER_IMAGE=auditbeat:latest \ 
  STACK_VERSION=8.12.0-SNAPSHOT docker-compose exec beat ls -l /usr/bin/tini

DOCKER_IMAGE=auditbeat:latest \
  STACK_VERSION=8.12.0-SNAPSHOT \
  docker-compose stop
```

image


image

File: dev-tools/packaging/templates/ironbank/auditbeat/Dockerfile [hunk 1]
```diff
@@ -47,6 +47,10 @@ COPY LICENSE /licenses/elastic-${ELASTIC_PRODUCT}
 # Add a dumb init process
 COPY tinit /tinit
 RUN chmod +x /tinit
+# Help with supporting the override in the ECK entrypoint
+# https://github.com/elastic/cloud-on-k8s/blob/272fd0f2b344b1f86f04decb561eceab8a5a3254/pkg/controller/agent/pod.go#L455
```
Comment (v1v) [inline dev-tools/packaging/templates/ironbank/auditbeat/Dockerfile:51]: I don't know whether this sentence makes sense only for the elastic-agent.

**Review by jlind23 — APPROVED** [review]
Thanks for this change, should we also prepare a PR for Elastic Agent 7.17.x and backport this one too?

**Comment by v1v** [comment]
> Thanks for this change, should we also prepare a PR for Elastic Agent 7.17.x and backport this one too?

I planned to backport this PR and then apply the changes for Elastic Agent 7.17.x changes in the same backported PR, so there will be one PR only for 7.17 with the additional changes in elastic agent, if that's ok with you

**Pull request merged**